Remove experimental_when_header feature (#6285)
apollographql/router: docs/source/reference/migration/from-router-v1.mdx

    metrics_reference_mode: standard
```

### Logging

If you used the `experimental_when_header` feature previously like this for example:

```yaml title="router.previous.yaml"
telemetry:
  exporters:
    logging:
      # If one of these headers matches we will log supergraph and subgraphs requests/responses
      experimental_when_header: # REMOVED
        - name: apollo-router-log-request
          value: my_client
          headers: true # default: false
          body: true # default: false
```

> This feature not longer exists and can be replaced with another one included in custom telemetry. Here is an example on how you could have the same behavior with the new configuration:

Reviewer: ```suggestion
This feature no longer exists and can be replaced with another one included in custom telemetry. Here is how you would configure custom telemetry to achieve the same result:
```
